Added Touch Behavior Migration Guide

Axemasta opened the pull request:
As discussed in the MCT discord, I have written a migration guide that can both help people migrating from XCT aswell as documenting how we can set the binding context on the behavior

This is my first time contributing to microsoft docs so I am very much open to advise & critique of my writing. I am used to writing technical documentation for internal use so I am not sure if my tone & examples, plus use of links is appropriate. Any & all critiques will be graciously accepted 😇

Axemasta commented:
@bijington I have raised the changed we discussed, when you get the time could you give these changes a review & let me know your thoughts 😃

bijington commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -267,3 +267,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+## Migrating From Xamarin Community Toolkit
+
+In the [Xamarin Community Toolkit](https://github.com/xamarin/XamarinCommunityToolkit), there was the [`TouchEffect`](https://learn.microsoft.com/en-us/dotnet/api/xamarin.communitytoolkit.effects.toucheffect?view=xamarin-community-toolkit-sdk) if you are upgrading an app from Forms to Maui there are a number of changes that you need to be aware of.
```
As we are on the Microsoft docs page we need to use the full names

```suggestion
In the [Xamarin Community Toolkit](https://github.com/xamarin/XamarinCommunityToolkit), there was the [`TouchEffect`](https://learn.microsoft.com/en-us/dotnet/api/xamarin.communitytoolkit.effects.toucheffect?view=xamarin-community-toolkit-sdk) if you are upgrading an app from Xamarin.Forms to .NET MAUI there are a number of changes that you need to be aware of.
```

Axemasta replied:
Fully qualified all names

bijington commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -267,3 +267,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+In the [Xamarin Community Toolkit](https://github.com/xamarin/XamarinCommunityToolkit), there was the [`TouchEffect`](https://learn.microsoft.com/en-us/dotnet/api/xamarin.communitytoolkit.effects.toucheffect?view=xamarin-community-toolkit-sdk) if you are upgrading an app from Forms to Maui there are a number of changes that you need to be aware of.
+
+The XCT implementation was an attached effect and was applied to an element in the following way:
```
```suggestion
The Xamarin Community Toolkit implementation was an attached effect and was applied to an element in the following way:
```

bijington commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -267,3 +267,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+You should be aware of the 2 following major changes to this effect:
+1. `TouchBehavior` is now a `PlatformBehavior`, not an `AttachedEffect`
+2. The `BindingContext is not automatically set
```
```suggestion
2. The `BindingContext` is not automatically set
```

jfversluis replied:
Can we maybe also write a little bit more about what that means exactly? More experienced developers will know, newcomers will maybe have trouble understanding what the symptoms are they might be seeing and what problem this solves.

Doesn't have to be super extensive, just something like: that means data-binding will not automatically work and pick up values in your view models.

I see you're expanding on this a little bit later? Always good to add a reference.

Axemasta replied:
I've turned this article into 2 sections and put links to each in this bullet list so it is clear about each point

bijington commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -267,3 +267,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+### Setting The TouchBehavior Binding Context
+
+As noted in the previous section, the `BindingContext` of the behavior is not set automatically, the reason for this is outlined in the Maui documentation for [Behaviors](https://learn.microsoft.com/en-gb/dotnet/maui/fundamentals/behaviors?view=net-maui-8.0#create-a-net-maui-behavior):
```
```suggestion
As noted in the previous section, the `BindingContext` of the behavior is not set automatically, the reason for this is outlined in the .NET MAUI documentation for [Behaviors](https://learn.microsoft.com/en-gb/dotnet/maui/fundamentals/behaviors?view=net-maui-8.0#create-a-net-maui-behavior):
```

bijington reviewed (commented):
@Axemasta the changes look like a great step forward! I've made some small suggestions but otherwise I think it is good.

jfversluis commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+## Migrating From Xamarin Community Toolkit
+
+In the [Xamarin Community Toolkit](https://github.com/xamarin/XamarinCommunityToolkit), there was the [`TouchEffect`](https://learn.microsoft.com/en-us/dotnet/api/xamarin.communitytoolkit.effects.toucheffect?view=xamarin-community-toolkit-sdk) if you are upgrading an app from Forms to Maui there are a number of changes that you need to be aware of.
```
```suggestion
In the [Xamarin Community Toolkit](https://github.com/xamarin/XamarinCommunityToolkit), there was the [`TouchEffect`](/dotnet/api/xamarin.communitytoolkit.effects.toucheffect?view=xamarin-community-toolkit-sdk) if you are upgrading an app from Forms to Maui there are a number of changes that you need to be aware of.
```

I think this should work and actually we'll get a warning otherwise to not do hard links.

As an aside: it's always good at the very least to remove the `en-us` identifier so the reader will be redirected to their own localized version if there is one.

jfversluis commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+### Migrating To Platform Behavior
+
+The behavior is now applied to the elements behavior collection, see [Platform Behaviors](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/behaviors?view=net-maui-8.0#platform-behaviors) for more information.
```
```suggestion
The behavior is now applied to the elements behavior collection, see [Platform Behaviors](/dotnet/maui/fundamentals/behaviors#platform-behaviors) for more information.
```

jfversluis commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+### Setting The TouchBehavior Binding Context
+
+As noted in the previous section, the `BindingContext` of the behavior is not set automatically, the reason for this is outlined in the Maui documentation for [Behaviors](https://learn.microsoft.com/en-gb/dotnet/maui/fundamentals/behaviors?view=net-maui-8.0#create-a-net-maui-behavior):
```
```suggestion
As noted in the previous section, the `BindingContext` of the behavior is not set automatically, the reason for this is outlined in the Maui documentation for [Behaviors](/dotnet/maui/fundamentals/behaviors#create-a-net-maui-behavior):
```

jfversluis commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+As noted in the previous section, the `BindingContext` of the behavior is not set automatically, the reason for this is outlined in the Maui documentation for [Behaviors](https://learn.microsoft.com/en-gb/dotnet/maui/fundamentals/behaviors?view=net-maui-8.0#create-a-net-maui-behavior):
+
+> .NET MAUI does not set the BindingContext of a behavior, because behaviors can be shared and applied to multiple controls through styles.
```
```suggestion
> .NET MAUI does not set the `BindingContext` of a behavior, because behaviors can be shared and applied to multiple controls through styles.
```

Axemasta replied:
Done

jfversluis commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,94 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
 You can find the source code for `TouchBehavior` over on the [.NET MAUI Community Toolkit GitHub repository](https://github.com/CommunityToolkit/Maui/blob/main/src/CommunityToolkit.Maui/Behaviors/TouchBehavior.shared.cs).
+
+
+## Migrating From Xamarin Community Toolkit
```
Not sure if it's relevant but should you include a full example?

This is how your Effect looked like in Xamarin, this is the same Behavior in .NET MAUI so you can really easily spot the differences? Or are they that small that this doesn't make sense?

Axemasta replied:
I'm happy to add an extra example, in my mind I put the xct example at the start of the example and then I have the equivalent maui example later. I'll put them together so its clearer for the reader :)

Axemasta replied:
I've merged the XCT & Maui examples into the same section so its clear that A migrates to B

jfversluis reviewed (changes requested):
Awesome that you're doing this! Couple of minor things (to consider). Thank you!

Axemasta commented:
I've restructured my writing based on feedback, the reviews are likely stale now.

I've made sure names are fully qualified & links are relative + not region / view locked.

Axemasta commented:
Based on the changes made in MCT, I've updated this article ([see PR](https://github.com/CommunityToolkit/Maui/pull/1791))

The `TouchBehavior` now has its `BindingContext` automatically set to the `VisualElement` it is attached to's `BindingContext`. This removed the need for the article to talk about referencing elements to grab their context.

Instead I've added a difference table between `TouchEffect` and `TouchBehavior`s api aswell as a coded example of the same feature on each platform!

Axemasta commented:
@bijington would I be able to get a 2nd review on these docs now please, MCT has merged the PR for the new Binding behavior so this migration guide will be accurate to whats in the next release 😄

Axemasta commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,151 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+In the Xamarin Community Toolkit the `TouchEffect` was implemented as an `AttachedEffect`. To use the effect you would use the attached properties and apply to any `VisualElement`
+
+In .NET Maui the `TouchBehavior` is implemented as a `PlatformBehavior` which is now applied to the elements behavior collection, see [Platform Behaviors](/dotnet/maui/fundamentals/behaviors#platform-behaviors) for more information.
```
Good spot, I've updated every instance of `Maui` to `MAUI`

Axemasta commented on docs/maui/behaviors/touch-behavior.md:
```diff
@@ -269,3 +269,151 @@ You can find an example of this behavior in action in the [.NET MAUI Community T
+
+In .NET Maui the `TouchBehavior` is implemented as a `PlatformBehavior` which is now applied to the elements behavior collection, see [Platform Behaviors](/dotnet/maui/fundamentals/behaviors#platform-behaviors) for more information.
+
+> Note: By default in .NET Maui `PlatformBehavior`s will not set a BindingContext, this is because Behaviors can be shared in styles. The `TouchBehavior` will set it's `BindingContext` equal to the `VisualElement` it is applied to. This means that you should not share the `TouchBehavior` between elements via styles.
```
Good spot, updated

Axemasta commented:
I've implemented @MartyIX's suggestions. I hopefully should have pushed a fix for the anchor links (i didn't check in the changes in my previous commit... 🤦‍♂️)

Unsure on how to fix the tabs as the validator is still complaining. I unfortunately don't have access to the link to the documentation it provides, I get redirected to a login page & if I continue it says I don't have access 😱

Axemasta commented:
Ok I have got the tabs stuff working... reverse engineering for the win 🏆
